components/esp_common/Kconfig

```diff
@@ -29,6 +29,13 @@ menu "Common ESP-related"
             which calls app_main(). If app_main() returns then this task is deleted
             and its stack memory is freed.
 
+    config ESP_MAIN_TASK_CORE_AFFINITY
+        int "Main task core affinity"
+        default 0
```

Comment:
Rather than using an `int` here, please use a `choice` to constrain this to the available CPU range. For an example of this pattern, see: https://github.com/espressif/esp-idf/blob/master/components/lwip/Kconfig#L595